app/src/main/java/com/rockthevote/grommet/ui/eventFlow/SessionProgressDialogFragment.java

```diff
@@ -63,13 +74,34 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         }
 
         ButterKnife.bind(this, view);
-        updateView();
-
         return view;
     }
 
-    public void updateView() {
-        // update count totals
+
+    private void observeData() {
+        viewModel.getSessionData().observe(this, data -> {
```

Comment:
You should use `getViewLifecycleOwner()` instead of `this` otherwise there's a potential for this to get out of sync in the lifecycle or interact with the view before it's ready.

Also for fragments, you're generally going to want to do this in `onViewCreated` or (now deprecated) `onActivityCreated`